config/config-api/src/com/thoughtworks/go/config/BasicCruiseConfig.java

```diff
@@ -1367,6 +1366,16 @@ public boolean canDeletePluggableSCMMaterial(SCM scmConfig) {
     }
 
     @Override
+    public void setPartials(List<PartialConfig> partials) {
+        this.partials = new Cloner().deepClone(partials);
```

Comment:
Partials contain other config objects such as  pipelineConfig internally which are merged with main config and validated. Validation errors (if any) will get added to the pipelineConfig object, which is later displayed in server-health messages. Need to clone so that the validation errors are not added to the cached-known-partials.